packages/suite/src/utils/suite/guide.ts

```diff
@@ -0,0 +1,38 @@
+import { GuideCategoryType, GuideItemType } from '@suite-types/guide';
+
+export const getItemTitle = (item: GuideItemType, language: string): string =>
+    item.title[language] || item.title.en;
+
+export const findParentCategories = (
+    item: GuideItemType,
+    root: GuideCategoryType,
+): GuideCategoryType[] | null => {
+    const itemParentId = item.id.split('/').slice(0, -1).join('/');
+
+    const findParentsRecursive = (
+        index: number,
+        item: GuideCategoryType,
+        searchedId: string,
+    ): GuideCategoryType[] | null => {
+        const currentItemId = `${searchedId
+            .split('/')
+            .slice(0, index + 1)
+            .join('/')}`;
+
+        const itemToLook = (item.children as any[]).find(i => i.id === currentItemId) as
+            | GuideCategoryType
+            | undefined;
+
+        if (!itemToLook) return null;
+
+        if (itemToLook.id === searchedId) return [itemToLook];
+
+        const foundParents = findParentsRecursive(index + 1, itemToLook, searchedId);
+
+        if (!foundParents) return null;
+
+        return [itemToLook].concat(foundParents);
+    };
+
+    return findParentsRecursive(1, root, itemParentId);
```

Comment:
I can think of some improvements for this function. If I get it right, it should accept a node ID and an index object and return an array of parent of the given id in tree-descending order. Eg. for `/cat/dog/mouse` it should return
```js
[
  {
    id: '/',
    ...
  },
  {
    id: 'cat'
    ...
  }
  {
    id: '/cat/dog'
    ...
  }
]
```

This get complicated as we need to traverse up the tree while the root object only has references going down the tree. However, since the given `id` contains the whole path, we can explode it and then traverse down the tree following the natural references. Something like:

```typescript
export const getAncestorIds = (
    id: string,
// btw: I think it's better to return an empty array if there are no parents rather than `null`.
): string[] => {
    return id
        .split('/')
        // omit the item itself - only consider its ancestors
        .slice(0, -1)
        // 'Absolutize' each exploded parent, effectively getting its ID.
        .reduce((ids, ancestor) => {
            const id = `${ids[ids.length - 1] || '/'}${ancestor}`;
            return ids.concat(id)
        }, []);
}

// We'll need this as a standalone function anyway (for the tooltips link feature)
function getNodeById(id: string, root: GuideCategoryType): GuideItemType {
    if (id === root.id) {
        return root;
    }

    return root.children?.map(child => {
        return getNodeById(id, child)
    }).find(it => it !== undefined)
}

export const findParentCategories = (item: GuideItemType, root: GuideCategoryType): GuideItemType[] => {
    const ancestorIds = getAncestorIds(item.id)
    return ancestorIds.map(id => getNodeById(id, root));
} 
```

Reply:
Thank you for your contribution, working great with small fixes